confirmation_rule.py

```diff
@@ -117,19 +112,24 @@ def find_latest_confirmed_descendant(store, latest_confirmed_root) -> Root:
     # starting with the child of the latest_confirmed_root
     # move towards the head in attempt to advance confirmed block
     # and stop when the first unconfirmed descendant is encountered
+    head = canonical_roots[-1]
```

Comment:
```suggestion
    head = get_head(store)
```

we can write it this way to avoid dependency on the `get_canonical_roots` function

Reply:
Yeah. I prefer it in this way.